docs/docs/external-volumes

```diff
@@ -0,0 +1,125 @@
+
+```json
+{
+  "id": "hello",
+  "instances": 1,
+  "cpus": 0.1,
+  "mem": 32,
+  "cmd": "while [ true ] ; do touch /tmp/test-rexray-volume/hello ; sleep 5 ; done",
+  "container": {
+    "type": "MESOS",
+    "volumes": [
+      {
+        "containerPath": "/tmp/test-rexray-volume",
+        "persistent": {
+          "size": 100,
+          "name": "jdef-test-vol",
+          "provider": "external",
+          "options": { "external/driver": "rexray" }
+          },
+        "mode": "RW"
+      }
+    ]
+  },
+  "upgradeStrategy": {
+    "minimumHealthCapacity": 0,
+    "maximumOverCapacity": 0
+  }
+}
+```
+
+In the `volumes` section of app definition above:
+
+- `containerPath` specifies the root path inside the volume for storing data. This path is configurable, but defaults to `data/`. See [the REX-Ray documentation on root paths](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#volume-root-path-031) for more information.
+
+- The `external/driver` options specifies which Docker volume driver we want to use for storage. If you are running Marathon on DCOS, this value should always be `rexray`. [Learn more about REX-ray](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/).
+
+- `name` is the name by which your Docker volume driver looks up your volume. When your task is staged on an agent, the volume driver queries the storage service for a volume with this name. If one does not exist, it's created. Otherwise, the existing volume is re-used.
+
+- Volumes will always be available under `$REXRAY_HOME/var/lib/rexray/volumes/$DVDI_VOLUME_NAME`. **Note:** This applies only to non-Docker apps; Docker apps can specify the mount point inside the app definition. See [the REX-Ray documentation on data directories](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#data-directories) for more information.
+
+### Using a Docker Container
+
+Below is a sample app definition that uses a Docker container and specifies an external volume:
+
+```json
+{
+  "id": "/test-docker",
+  "instances": 1,
+  "cpus": 0.1,
+  "mem": 32,
+  "cmd": "/usr/bin/tail -f /dev/null",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "alpine:3.1",
+      "network": "HOST",
+      "forcePullImage": true
+    },
+    "volumes": [
+      {
+        "containerPath": "/data/test-rexray-volume",
+        "persistent": {
+          "size": 100,
+          "name": "test",
+          "provider": "external",
+          "options": { "external/driver": "rexray" } 
+        },
+        "mode": "RW"
+      }
+    ]
+  },
+  "upgradeStrategy": {
+    "minimumHealthCapacity": 0,
+    "maximumOverCapacity": 0
+  }
+}
+```
+
+For more information, refer to the [REX-Ray documentation](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/#docker-containerizer-with-marathon).
+
+## Potential Pitfalls
+
+- Your storage provider may have particular limitations. For instance, on Amazon's EBS, you can only assign one task per volume.
```

Comment:
currently we're enforcing this for all external volume providers, not just DVDI using rexray